x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -262,7 +264,16 @@ public void authorizeIndexAction(RequestInfo requestInfo, AuthorizationInfo auth
                 // index and if they cannot, we can fail the request early before we allow the execution of the action and in
                 // turn the shard actions
                 if (SearchScrollAction.NAME.equals(action)) {
-                    authorizeIndexActionName(action, authorizationInfo, null, listener);
+                    try {
```

Comment:
As you've mentioned the try-catch structure unfortunately doesn't fit too nicely with the async listener idiom .
I personally would've maintained a check like `request instanceof SearchScrollRequest && ((SearchScrollRequest)request).scrollId() != null` as I'm not a fan of catching runtime exceptions . 

Again, this is a feedback with my perspective, the codebase is cot consistent in this regard; it's your choice.

Reply:
I changed it to use `ActionRunnable.supply(...).run()`, which takes a `CheckedSupplier` to handle the exception. Honestly, the code is less readable with this change. But it is more in line with the overwhelming callback style.